demos/http/http_demo_s3_download/demo_config.h

```diff
@@ -70,26 +79,68 @@
  *
  * @note This certificate should be PEM-encoded.
  */
-#ifndef ROOT_CA_CERT_PATH
-    #define ROOT_CA_CERT_PATH    "certificates/BaltimoreCyberTrustRoot.crt"
+#ifndef ROOT_CA_CERT_PATH_S3
+    #define ROOT_CA_CERT_PATH_S3    "certificates/BaltimoreCyberTrustRoot.crt"
 #endif
 
 /**
- * @brief The pre-signed GET URL generated by the python script located in
- * common/src/presigned_urls_gen.py
+ * @brief Path of the file containing the client certificate for TLS
+ * authentication with AWS IOT credential provider.
  *
- * @note This script requires AWS CLI to be configured. For instructions, see
- * https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-configure.html
+ * @note This certificate should be PEM-encoded.
+ */
+#ifndef CLIENT_CERT_PATH
+    #define CLIENT_CERT_PATH    "...insert here..."
+#endif
+
+/**
+ * @brief Path of the file containing the client private key for TLS
+ * authentication with AWS IOT credential provider.
  *
- * Run this script and paste the output S3_PRESIGNED_GET_URL below.
+ * @note This key should be PEM-encoded.
+ */
+#ifndef CLIENT_PRIVATE_KEY_PATH
+    #define CLIENT_PRIVATE_KEY_PATH    "...insert here..."
+#endif
+
+/**
+ * @brief Define AWS IOT thing name.
+ */
+#define AWS_IOT_THING_NAME                      "...insert here..."
+
+/**
+ * @brief Endpoint for the AWS IOT credential provider.
  *
- * #define S3_PRESIGNED_GET_URL         "...insert here..."
+ * @note Can be found with
+ * `aws iot describe-endpoint --endpoint-type iot:CredentialProvider` from
+ * the AWS CLI.
+ */
+#define AWS_IOT_CREDENTIAL_PROVIDER_ENDPOINT    "...insert here..."
+
+/**
+ * @brief Role alias for accessing the credential provider.
```

Comment:
More information about an example for the IAM Role will be helpful for the customer

Reply:
README for the demo contains all the information on how to create role alias for IAM Role. Also I have added some more info in the comment. Let me know if anything else needs to be added.